### PR TITLE
Remove MONO_ASMCTX_REFONLY

### PR DESCRIPTION
**Pull request by lambdageek**

### CODE REVIEW

File: src/mono/mono/metadata/appdomain.c [hunk 1]
```diff
@@ -1097,6 +1097,7 @@ mono_domain_assembly_preload (MonoAssemblyLoadContext *alc,
 
 	g_assert (alc);
 	g_assert (domain == mono_domain_get ());
+        g_assert (!refonly);
```
Comment (vargaz) [inline src/mono/mono/metadata/appdomain.c:1101]: Indentation

**Review by vargaz — APPROVED** [review]

File: src/mono/mono/metadata/assembly.c [hunk 2]
```diff
@@ -1475,19 +1475,19 @@ struct AssemblySearchHook {
...
 
 static MonoAssembly*
-mono_assembly_invoke_search_hook_internal (MonoAssemblyLoadContext *alc, MonoAssembly *requesting, MonoAssemblyName *aname, gboolean refonly, gboolean postload)
+mono_assembly_invoke_search_hook_internal (MonoAssemblyLoadContext *alc, MonoAssembly *requesting, MonoAssemblyName *aname, gboolean postload)
 {
 	AssemblySearchHook *hook;
 
 	for (hook = assembly_search_hook; hook; hook = hook->next) {
-		if ((hook->refonly == refonly) && (hook->postload == postload)) {
+		if ((hook->refonly == FALSE) && (hook->postload == postload)) {
```
Comment (CoffeeFlux) [inline src/mono/mono/metadata/assembly.c:1483]: Can't you remove the `refonly` check entirely?

File: src/mono/mono/metadata/assembly.c [hunk 3]
```diff
@@ -2038,7 +2043,7 @@ mono_assembly_request_open (const char *filename, const MonoAssemblyOpenRequest
 			return NULL;
 		}
 
-		if (load_req.asmctx != MONO_ASMCTX_REFONLY) {
```
Comment (CoffeeFlux) [inline src/mono/mono/metadata/assembly.c:2041]: Is there any reason to leave this block?

File: src/mono/mono/metadata/assembly.c [hunk 4]
```diff
@@ -2430,7 +2437,7 @@ mono_assembly_request_load_from (MonoImage *image, const char *fname,
 	 * this image and we won't be able to look for a different
 	 * candidate. */
 
-	if (asmctx != MONO_ASMCTX_REFONLY) {
+	{
```
Comment (CoffeeFlux) [inline src/mono/mono/metadata/assembly.c:2440]: Same comment re: block

**Review by CoffeeFlux — APPROVED** [review]
I assume we also want to remove the argument from `mono_image_open_from_data_internal ` eventually? Other than feedback LGTM.

File: src/mono/mono/metadata/assembly.c (hunk 3 above)
Comment (lambdageek) [reply src/mono/mono/metadata/assembly.c:2041]: *shrug* wanted to leave the indentation unchanged

File: src/mono/mono/metadata/assembly.c (hunk 2 above)
Comment (lambdageek) [reply src/mono/mono/metadata/assembly.c:1483]: you can still install refonly search hooks.

**Pull request merged**